llvm/lib/IR/Verifier.cpp

```diff
@@ -5235,6 +5235,28 @@ void Verifier::visitIntrinsicCall(Intrinsic::ID ID, CallBase &Call) {
     }
     break;
   }
+  case Intrinsic::ucmp:
+  case Intrinsic::scmp: {
+    Type *SrcTy = Call.getOperand(0)->getType();
+    Type *DestTy = Call.getType();
+
+    Check(DestTy->getScalarSizeInBits() >= 2,
+          "result type must be at least 2 bits wide", Call);
+
+    bool IsDestTypeVector = DestTy->isVectorTy();
+    Check(SrcTy->isVectorTy() == IsDestTypeVector,
+          "[us]cmp source and destination must both be a vector or neither",
```

Comment:
Small nit: I feel like `ucmp/scmp` would be easier to read than `[us]cmp`.

Also, I don't think I would use the word destination here as there is no exact "destination" per se, so something like `[us]cmp argument and result types must both be either vector or scalar types` would be better.